src/containers/StructurePage/folderComponents/SettingsMenu.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2016-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React from 'react';
+import Button from '@ndla/button';
+import { Settings } from '@ndla/icons/editor';
+import styled from '@emotion/styled';
+import { useState } from 'react';
+import Overlay from '../../../components/Overlay';
+import RoundIcon from '../../../components/RoundIcon';
+import {
+  SubjectTopic,
+  TaxonomyElement,
+  TaxonomyMetadata,
+} from '../../../modules/taxonomy/taxonomyApiInterfaces';
+import SettingsMenuDropdown from './SettingsMenuDropdown';
+import { PathArray } from '../../../util/retrieveBreadCrumbs';
+
+const StyledDivWrapper = styled('div')`
+  position: relative;
+  display: flex;
+
+  > button {
+    outline: none;
+  }
+`;
+
+interface Props {
+  type: string;
+  setShowAlertModal: (show: boolean) => void;
+  id: string;
+  name: string;
+  path: string;
+  showAllOptions: boolean;
+  metadata: TaxonomyMetadata;
+  locale: string;
+  getAllSubjects: () => Promise<void>;
+  refreshTopics: () => Promise<void>;
+  subjectId: string;
+  setResourcesUpdated: (updated: boolean) => void;
+  saveSubjectItems: (
+    subjectid: string,
+    saveItems: { topics?: SubjectTopic[]; loading?: boolean; metadata?: TaxonomyMetadata },
+  ) => void;
+  saveSubjectTopicItems: (topicId: string, saveItems: Pick<TaxonomyElement, 'metadata'>) => void;
+  parent: string;
+  structure: PathArray;
+}
+
+const SettingsMenu = ({
+  type,
+  setShowAlertModal,
+  id,
+  name,
+  path,
+  locale,
+  metadata,
+  showAllOptions,
+  getAllSubjects,
+  refreshTopics,
+  subjectId,
+  setResourcesUpdated,
+  saveSubjectItems,
+  saveSubjectTopicItems,
+  parent,
+  structure,
+}: Props) => {
+  const [open, setOpen] = useState(false);
+  const [editMode, setEditMode] = useState('');
```

Comment:
Fins det flere editModes enn to? Kunne kanskje ha vært en boolean. Om det fins flere, kanskje gjøre om til enum.

Reply:
Det finnes en god del. En enum kunne forsåvidt vært lurt.